lib/erl_docgen/src/eep48_to_markdown.erl

```diff
@@ -144,19 +147,52 @@ convert(File, Docs) ->
           fun(MFA1, MFA2) ->
                   erl_anno:line(element(2, MFA1)) >= erl_anno:line(element(2, MFA2))
           end, Docs),
-    convert(string:split(File, "\n", all), [], SortedDocs).
-convert(File, Acc, []) ->
-    unicode:characters_to_binary([[A,$\n] || A <- File ++ Acc]);
-convert(File, Acc, [{_, Anno, _, #{ <<"en">> := D }, _} | T]) ->
-    {Before, After} = lists:split(erl_anno:line(Anno)-1, File),
-    convert(Before, [comment(render_docs(D, init_config(D, #{})))|After] ++ Acc, T).
+    Lines = string:split(File, "\n", all),
+    {ModuleHeader, ModuleBody} =
+        lists:foldl(fun document_module/2, {Lines, []}, SortedDocs),
+    ModuleHeader ++ ModuleBody.
```

Comment:
I did not mean to rewrite something that already works, like your implementation (I can remove the last commit).
However, I would like to understand if we favour recursion over maps and folds.

For me, recursion is extremely useful if I need to exit early from a function, but if I am going to traverse the whole list, I tend to think that maps and folds are more clean and remove the burden of figuring out which argument is really state used in the function and which one is simply the output argument.

Reply:
I too prefer to use the map/fold functions if possible. However when quickly prototyping something I tend to not think too much about what I use, and in this case this function at first was neither a map nor a fold, but as I continued work it turned into a fold. I just didn't do the final refactor/cleanup yet :)

Reply:
Great, just to see if you have any conventions :)